backend/de.metas.business/src/main/java/de/metas/project/budget/BudgetProjectResourceRepository.java

```diff
@@ -193,4 +209,32 @@ public InSetPredicate<ProjectId> getProjectIdsPredicateByResourceGroupIds(
 		final ImmutableSet<ProjectId> projectIdsEffective = ProjectId.ofRepoIds(projectRepoIdsEffective);
 		return InSetPredicate.only(projectIdsEffective);
 	}
+
+	@NonNull
+	public BudgetProjectResource save(
```

Comment:
wrote about this somewhere above. consider implementing createOrUpdateAll(List<UpsertBudgetProjectResourceRequest> requests, ProjectId projected)